### PR TITLE
Features : Weather app

### PR DESCRIPTION
**Pull request by Fanger53**
# Weather App
 this app consist use an API to get information about the weather from a specific place
## in this project, I use the next technologies : 
- Html
- Scss
- javascript
- Webpack
- API from open weather map

### CODE REVIEW

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### **STATUS: CHANGES REQUESTED🛑🛑🛑🛑**

Hello @Fanger53 
Great job on this project. I can tell you put it in a lot of work. 
Great job on the background gradient.

However, some changes are requested to get it approved.

[REQUIRED]: Kindly add `"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  }
` to package.json. This will allow you to run commands like npm start to get the app running.

[REQUIRED]: In weather.js, kindly use arrow function.

[REQUIRED]: Add a toggle button to change from celsius to fahrenheit

[REQUIRED]: The UI changes its appearance based on the received data.

[OPTIONAL]: it fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location  

[OPTIONAL]: it has a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API

That's it.🙏

Make the requested changes and ask for another review👍👌

Happy coding💻💻💻

**Comment by Fanger53** [comment]
its a nice feature but it's not necessary I like how looks de Celsius and Fahrenheit at the same time 
[REQUIRED]: Add a toggle button to change from celsius to Fahrenheit THIS is optional is not a requirement

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  -  Follows gitwork-flow. 💯
  -  App is functional and available online. 💯 

- ### To Improve
  
  - [ ] [MANDATORY] As the previous reviewer mentioned, you must have a toggle control to determine if the result will be shown in Celsius or Fahrenheit ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
  - [ ] [MANDATORY] It must change its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
      - Example: if the weather is cloudy or rainy, it should change the background accordingly.

  NOTE: Those features are **MANDATORY**!    
  
_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

File: src/modules/weather.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export default class Weather {
+  constructor(city) {
+    this.apiKey = '88dd50190cb7c4a2c8e55d61d1341d41';
+    this.city = city;
+  }
+
+  getWeather = async () => {
+    const apiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${this.city}&units=metric&appid=${this.apiKey}`;
+    const response = await fetch(apiUrl, { mode: 'cors' });
```
Comment (Forison) [inline src/modules/weather.js:9]: * Let us use error handling called `try and catch` block, based on this we can display error on the UI when the data fetching fails as mention above.

**Review by Forison — APPROVED** [review]
# Status::Approved✔️ 

#### Good job on addressing the issue raised in the previous review, the application changes appearance based on result and we are able to view temperature in both required units😇.

**[Optional because it wasn't mentioned earlier, but very important and a mandatory requirement]**

* [ ] It has error handling for `promises` or `async/await` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
Let's display an error to the user when the wrong city or country in entered in the search field, currently errors are logged in the console but not information is seen by the user on the front-end. Check associated comments below👇.

**[Optional]**

- [ ]  it has a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))

 - [ ] it fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location
#### Happy coding

Addo Forison

**Pull request merged**